pkg/kernel/common.go

```diff
@@ -45,6 +45,16 @@ func Release() (string, error) {
 	return int8SliceToString(uname.Release[:]), nil
 }
 
+// Machine fetches the machine string of the current running kernel.
+func Machine() (string, error) {
+	var uname syscall.Utsname
+	if err := syscall.Uname(&uname); err != nil {
```

Comment:
potential perf nit (to be checked in profiles): We are potentially calling this syscall a bunch of times (2*_n_). We could potentially just call it once and get that result for both `Release()`, `Machine()`, etc

Reply:
This is cached on the caller side. https://github.com/parca-dev/parca-agent/blob/add_arch_metadata_label/pkg/metadata/system.go#L43